ide/db/src/org/netbeans/modules/db/explorer/dlg/CreateTableDialog.java

```diff
@@ -471,7 +471,7 @@ public DataTable(TableModel model) {
                 Map cmap = ColumnItem.getColumnProperty(i);
                 col.setIdentifier(cmap.get("name")); //NOI18N
                 columnName = NbBundle.getMessage (CreateTableDialog.class, "CreateTable_" + i); //NOI18N
-                columnWidth = (new Double(getFontMetrics(getFont()).getStringBounds(columnName, getGraphics()).getWidth())).intValue() + 20;
+                columnWidth = Double.valueOf(getFontMetrics(getFont()).getStringBounds(columnName, getGraphics()).getWidth()).intValue() + 20;
```

Comment:
Would it make sense to convert the `Double.valueOf(/*double yielding expression*/).intValue()` to `(int) /*double yielding expression*/`?

The above would then become:

```suggestion
                columnWidth = ((int) getFontMetrics(getFont()).getStringBounds(columnName, getGraphics()).getWidth()) + 20;
```

Reply:
I looked at the JLS and I'm going to leave it as is if you are okay with this.

The reason is is the JLS section 5.1.3 Narrowing Primitive Conversions talks about how valueOf() gets handled. The JLS section 5.5. Casting Contexts handles type casting and it's not really clear both narrow/expand the cast the same way. I would really have to look at this in great detail and the work just didn't justify the brain power. 8-)

So I decided to jjust go the route of guarantee the same behavior. There is a combination of narrowing and expanding here and it it's probably better to just guarantee the same behavior.

Reply:
this code is fairly harmless and is only calculating a column width for UI. Should be save to change like @matthiasblaesing suggested.

its essentially the width of a Rectangle2D, casted down to int + 20, thats it.

Reply:
Seems to be unchanged, maybe you forgot to add this?